R/load.config.R

```diff
@@ -1,74 +1,148 @@
+library("pryr")
+library("utils")
+
 #' Loads a yaml config file
 #' @param project A project (use default config file names for this project)
 #' @param sp Subproject to activate
 #' @param file file path to config file, allows you to specify an exact file.
 #' @export
-load.config = function(project=NULL, sp=NULL, filename=NULL) {
-	if ( ! requireNamespace("yaml", quietly=TRUE)) {
-		warning("Package yaml is required to load yaml config files.")
-		return
-	}
+load.config = function(project=NULL, sp = NULL, filename=NULL, uses_paths_section = FALSE) {
+
+	# Derive project folder from environment variables and project name.
 	if (is.null(project)) { 
-		projectDir = options("PROJECT.DIR")
+		project_dir = options("PROJECT.DIR")
 	} else {
-		projectDir = paste0(Sys.getenv("CODEBASE"), project)
-	}
-	# If no file is specified, try these default locations
-	yamls = list("metadata/config.yaml",
-						"metadata/project_config.yaml",
-						paste0("metadata/", project, ".yaml"))
-	cfg = NULL
-	if (! is.null(filename)) {
-		yamls = c(filename, yamls)
+		codepath = Sys.getenv("CODE")
+		project_dir = if (is.null(codepath) | identical("", codepath)) project else file.path(codepath, project)
```

Comment:
OK, I was trying to follow the naming conventions here http://adv-r.had.co.nz/Style.html, but Google agrees more w/ camel for normal variables. It actually even more prefers the (awful, IMO) dots-as-delimiters style for variable naming: https://google.github.io/styleguide/Rguide.xml#identifiers
I'm OK w/ either, and actually slightly prefer the camel case way that it sounds (and looks) like is already in use most places. I just went the other route because it was by Wickham & this was by Google. I'm totally on board w/ camel casing, though. If it's not already so, can I make this official somewhere, maybe in `labDocs`?